### ISSUE TITLE
add typescript defintions for $sentry

### ISSUE THREAD

**Issue by simllll** [nuxt-community/sentry-module#67]
### What problem does this feature solve?

right now $sentry is not available in the context or in a component. therefore no typescript definition is associated when using this.$sentry or in any other context.

### What does the proposed changes look like?

To solve this a proper typescript defintion is needed, right now to workaround this issue you simple can do:
1.) create a new "index.d.ts" file
2.) paste:

declare module 'vue/types/vue' {
    interface Vue {
        $sentry: any;
}
}
see also // https://vuejs.org/v2/guide/typescript.html#Augmenting-Types-for-Use-with-Plugins

but the correct solution would be not using "any", but the correct type. I tried it with:
import { NodeClient } from '@sentry/node';

but this collides with some other things, and actually it would be wrong on the client side too.

Could be the right way to provide a "unified" interface for both client&server and wrap the actual sentry calls behind it?

<div align="right"><sub><em>This feature request is available on <a href="https://cmty.app/nuxt">Nuxt</a> community (<a href="https://cmty.app/nuxt/sentry-module/issues/c51">#c51</a>)</em></sub></div>

**Comment by kevinmarrec** [comment]
Hi @simllll .

Indeed there is a need to define such typescript definitions so you get proper autocomplete & type checking. Declaring the type as `any` is also indeed the right way to go unless it gets properly defined.

Furthermore, you're again right about the fact both Server & Client seem to have different types.
Sentry instance in server comes from here : https://github.com/nuxt-community/sentry-module/blob/master/lib/module.js#L2, meanwhile Sentry instance in client comes from here : https://github.com/nuxt-community/sentry-module/blob/master/lib/sentry.client.js#L2.

Assuming `@sentry/node` source code https://github.com/getsentry/sentry-javascript/blob/master/packages/node/src/index.ts & how it gets implemented, it seems that on the server side of a Nuxt app, `$sentry` would be the whole package with all export.

It also seems that Sentry packages are individually exporting types : https://github.com/getsentry/sentry-javascript/blob/master/packages/browser/src/index.ts#L1

@simllll Did you figure out maybe how can we grab the types ?

We need to first know how to get the types and how they're supposed to be used Server & Client sides, then we' should be able to define the type of `$sentry` properly.

**Comment by rchl** [comment]
Client object is defined here https://github.com/getsentry/sentry-javascript/blob/master/packages/types/src/client.ts

I guess those types should work for both client and server?

**Comment by simllll** [comment]
As it turned out, they are not:
https://github.com/getsentry/sentry-javascript/issues/2158

I'm on it :)

Regards

**Comment by icehaunter** [comment]
Sorry to be a bother, but any progress on that? Would be very useful!

**Comment by icehaunter** [comment]
This works properly FYI:
```typescript
declare module 'sentry' {
  import { default as _Vue } from 'vue'

  import * as Sentry from '@sentry/browser'

  module 'vue/types/vue' {
    interface Vue {
      $sentry: typeof Sentry
    }
  }
}
```

**Comment by P4sca1** [comment]
This is the content of my `types/index.d.ts`:

```
import sentry from '@sentry/browser'

declare module 'vuex' {
  interface Store<S> {
    $sentry: typeof sentry
  }
}

declare module '@nuxt/types' {
  interface NuxtAppOptions {
    $sentry: typeof sentry
  }
}

declare module 'vue/types/vue' {
  interface Vue {
    $sentry: typeof sentry
  }
}

```

**Comment by rchl** [comment]
Resolved by #84
